### ISSUE TITLE
Request - private curatorial notations for records

### ISSUE THREAD

**Issue by mkoo** [ArctosDB/arctos#8338]
This issue originated in https://github.com/ArctosDB/arctos/issues/3536 and became stale-- after discussions, here's the refreshed summary:

Need: a means to allow curatorial staff to leave private (non-public) record level notes about condition or preservation need especially in the art collections. 

Options considered included:
1.  two new encumbrances: mask part attribute condition report and mask part attribute preservation need
2.  new private record attributes 
3.  new curatorial remarks that are always private

Main barrier/consideration is resource costs-- encumbrances are 'expensive' and slows exports down but they are essential. However if we can avoid unnecessary encumbrances then we all benefit. 

Last time the AWG discussed this, we were favoring option 3 (if I'm not mistaken!)
![Firefox_Screenshot_2024-11-26T05-37-04 990Z](https://github.com/user-attachments/assets/22e61241-3830-45e7-879c-d9ccfe6ff1aa)

**Comment by happiah-madson** [comment]
Strong preference for #3.

**Comment by dustymc** [comment]
I made https://github.com/ArctosDB/dev/issues/118 for this, but I'm now questioning that.


> encumbrances are 'expensive' and slows exports down

Clarification: As currently implemented encumbrances slow **everything** down. https://github.com/ArctosDB/dev/issues/118 will still have computational costs, but they should be much more manageable (and understandable) under such a system.

I was also going to suggest we develop scalable policy before this is implemented. If this works like I think it should, Phase Two will involve record attributes,  of which there are [very many](https://github.com/orgs/ArctosDB/discussions/6179). The "full" implementation of this would simply double that - we'd have a public 'tail condition' and a private 'tail condition' (the existence of which suggests we're already long-overdue for more-developed policies regarding what is an appropriate attribute), which seems neither sustainable nor usable.

Now I'm wondering if we can actually implement this entirely against record attributes, or if parts (and presumably other kinds) are necessary? The code isn't written and I'm guessing to some extent, but running 'private types' against one table is probably well under half the computational expense of two, but is also simpler to use and manage (restricted data go in one place, easy-peasey, nowhere to get lost) and possibly sufficient to cover the critical use cases. Would having even part-related things under one (or a very few) private record attribute(s) be sufficient, or is attaching that information to specific parts truly critical?

Adding to AWG agenda for further clarification, "as simple as possible, as complex as necessary" seems particularly relevant here.

**Comment by happiah-madson** [comment]
At the risk of opening cans of worms, has there been discussion about both record level and part level curatorial remarks? I've just been speaking with @dandistel who is contemplating encumbering all remarks because some of them are more curatorial than public.

**Comment by happiah-madson** [comment]
> That's why we're here, do we REALLY need both or can record-level sufficiently cover the bases

I think my comment about both is that when migrating, we had specimen (record) and DNA or sample (part) remarks that are things like "oh, we found this in a not so great place, salvaged it, and now it's over here" or "I assume the elution volume was 200 uL because I forgot to write it down in my lab notebook so here's hoping." Because these remarks were never public in our last database, they are written super colloquially and perhaps would have been written differently if they were public.

>eg would cataloged_item.private_remark and specimen_part.private_remark do whats needed

From my view, yes, this would do what is needed.

> otherwise it's probably the public's information, they paid for it, why make it difficult for them to access it??

For us, it's probably just tone. It's editorial rather than factual information that migrated over in remarks.

>Arctos has no security team

Understood!

**Comment by campmlc** [comment]
I think this would work for us as well: cataloged_item.private_remark and specimen_part.private_remark
We would like to be able to record a private part remark such as "oops, this vial of tissue was discovered under an ultralow freezer and obviously fell out at some point for some extended period of time and is now very rotten" at the same time as saying "condition = poor" in the public part remarks. I can also see use for private remarks at the cataloged item level, e.g this specimen had its tail pulled off by a group of kindergardeners on a school tour and it's been glued back on - we need to restrict future handling etc. I assume the art and cultural collections have even more issues with needing private remarks fields.

**Comment by dustymc** [comment]
>this would work for us as well: cataloged_item.private_remark and specimen_part.private_remark

Please confirm that you mean as a way to handle https://arctos.database.museum/info/ctDocumentation.cfm?table=ctencumbrance_action and an alternative to https://github.com/ArctosDB/dev/issues/118.

**Comment by campmlc** [comment]
If I understand correctly, we would not be using encumbrances for this, but rather using public/private flags or specifically designated private remarks fields for part and record attributes. This is what I thought I was supporting.
https://github.com/ArctosDB/dev/issues/118

**Comment by happiah-madson** [comment]
>Please confirm

I do. But I do think the fields could be more general, ie:

cataloged_item.private_remark and cataloged_item_part.private_remark

or 

record.private_remark and record_part.private_remark

**Comment by dustymc** [comment]
>not be using encumbrances for this

I'm trying to find a sustainable replacement for encumbrances (possibly one which does some things have have been requested for years - like allowing data to be born encumbered - and maybe simplifies some complexity, behind which people clearly get lost from time to time). https://github.com/ArctosDB/dev/issues/118 is a move, not an addition. Introducing some even-simpler always-restricted field would only be relevant here if it's a replacement for some existing encumbrance.

(Bigger-picture, I'm struggling with the usual: Are things they way they are because users were trying to shoehorn various data into the structure as they understood it? - in which case a new always-private field or three might be just the ticket - or does the existing structure do precisely what needs to be done? - in which case https://github.com/ArctosDB/dev/issues/118 is probably the correct first step towards a model we can afford.)

>record.private_remark and record_part.private_remark

That's ``table.field`` terminology but those aren't tables so ???? (I think we're on the same page anyway!)

**Comment by happiah-madson** [comment]
> that's table.field terminology but those aren't tables so ???? (I think we're on the same page anyway!)

😆😆😆😆😆😆😆😆😆😆😆😆 whooooopseeeeeeeeeee (I agree that we are on the same page)

**Comment by AJLinn** [comment]
>  I assume the art and cultural collections have even more issues with needing private remarks fields.

True! I support a "born private" or "born encumbered" by whatever mechanism works best on the back side. It's true we do get asked to make certain cultural information not-public by tribes or donor families, or by professional ethics I choose to not put information from the paper files into the Arctos record because I can't mask it.

>  Introducing some even-simpler always-restricted field would only be relevant here if it's a replacement for some existing encumbrance.

I have a "mask specimen remarks" encumbrance that I currently have only applied to three records. (I think I forgot I have it!) I'd use this private remarks field to move such comments over that are potentially culturally sensitive; or other remarks I'd like for internal use (e.g., we often make notes for updating certain fields by writing "Arctos: collector names are xxx" so we can search for that phrase as instructions for record updates).
<img width="1309" alt="Screenshot 2025-01-01 at 3 07 48 PM" src="https://github.com/user-attachments/assets/a84410e5-9868-43ba-933c-8d7a42612676" />

Others appear to be using this encumbrance extensively:
<img width="1369" alt="Screenshot 2025-01-01 at 3 25 40 PM" src="https://github.com/user-attachments/assets/0c9c4576-295b-42c1-9b3d-544ed9dbcd52" />

**Comment by dustymc** [comment]
Sending back to AWG for discussion re: https://github.com/ArctosDB/arctos/issues/8338#issuecomment-2501603302 -  https://github.com/ArctosDB/dev/issues/118 is implemented, seems to be working well, could serve as a model for record attributes, or possibly something even simpler could meet the needs.

**Comment by mkoo** [comment]
Work in Progress-- this shouldnt be controversial in implementation and will fulfill the functional need!
thanks for proceeding

**Comment by dustymc** [comment]
Recast as https://github.com/ArctosDB/arctos/issues/8589
